### PR TITLE
(DOCSP-8759): Port Auth Providers Overview

### PR DESCRIPTION
**Pull request by cbush**
- Renamed stitch -> Realm
- Reworked intro paragraph a bit
- Reworked Anonymous auth a bit

## Staged
https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker/port-auth-overview/authentication/providers

## Original
https://docs.mongodb.com/stitch/authentication/providers/

## JIRA
https://jira.mongodb.org/browse/DOCSP-8759

### CODE REVIEW

File: source/authentication/providers.txt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+You can use a single provider if you want all users to
+authenticate in the same way, or you can enable multiple
+providers for more flexibility. You can also link a user
+account from one provider to a user account from another
+provider by using the the client SDKs. For more information
+on linking, see :doc:`Link User Accounts
+</authentication/link>`.
+
+An example of an app that would benefit from multiple
```
Comment (mongomoe) [inline source/authentication/providers.txt]: Great example

File: source/authentication/providers.txt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+Realm provides several authentication providers that you can
+integrate into a client application to allow users to log in
+to your app.
+
+- For applications where you want users to be able to view or manipulate
+  data without registering or creating an account, you can use
+  :doc:`Anonymous Authentication </authentication/anonymous>`.
+
+- For applications where end users create an account or log in with
```
Comment (mongomoe) [inline source/authentication/providers.txt]: Switch to "end-users"

"For applications where end users create ..." 

should be:

"For applications where end-users create ..."

File: source/authentication/providers.txt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  existing credentials, you can use providers that integrate with
+  existing login services (:doc:`Facebook </authentication/facebook>`
+  and :doc:`Google </authentication/google>`), or use providers that
+  allow you or your end users to create new credentials
```
Comment (mongomoe) [inline source/authentication/providers.txt]: "end-users"

File: source/authentication/providers.txt [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  (:doc:`Email/Password </authentication/userpass>`, :doc:`API Keys
+  </authentication/api-key>`, :doc:`Custom JWT Auth
+  </authentication/custom-token>` and :doc:`Custom Function Auth
+  </authentication/custom-function>`).
+
+You can use a single provider if you want all users to
+authenticate in the same way, or you can enable multiple
+providers for more flexibility. You can also link a user
+account from one provider to a user account from another
+provider by using the the client SDKs. For more information
```
Comment (mongomoe) [inline source/authentication/providers.txt]: extra "the" after "provider by using"

"You can also link a user account from one provider to a user account from another provider by using the the client SDKs." 

should be:

"You can also link a user account from one provider to a user account from another provider by using the client SDKs."

File: source/authentication/providers.txt (hunk 2 above)
Comment (cbush) [reply source/authentication/providers.txt]: When used as a noun, "end user" is usually unhyphenated.

Comment (mongomoe) [reply source/authentication/providers.txt]: Sounds good!

**Pull request merged**